lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -608,20 +637,42 @@
         return toReturn;
     }());
 
+    // mappedDefaultLocale will get the default locale and update any deprecated
+    // collation/sort order values it may use
     var mappedDefaultLocale = function () {
-        var parts = platform.builtInRegexMatch(GetDefaultLocale(), /([^_]*)_?(.+)?/);
-        var locale = parts[1];
-        var collation = parts[2];
-        var availableBcpTags = reverseLocaleAcceptingCollationValues[locale];
+        let locale = undefined;
+        let collation = undefined;
+        if (isICU) {
+            // ICU's getDefaultLocale() will return a valid BCP-47/RFC 5646 langtag
+            locale = platform.getDefaultLocale();
+            const match = platform.builtInRegexMatch(locale, /-u(-[^\-][^\-]?-[^\-]+)*(-co-[^\-]+).*/);
+            if (match) {
+                collation = callInstanceFunc(StringInstanceReplace, match[2], "-co-", "");
+                locale = callInstanceFunc(StringInstanceReplace, locale, match[2], "");
+            }
+        } else {
+            // Windows' getDefaultLocale() will return a weird RFC4646 langtag
+            const parts = platform.builtInRegexMatch(platform.getDefaultLocale(), /([^_]*)_?(.+)?/);
+            locale = parts[1];
+            collation = parts[2];
+        }
 
-        if (collation === undefined || availableBcpTags === undefined) { return locale; }
+        if (collation === undefined) {
```

Comment:
Does removing `availableBcpTags === undefined` apply in general to ICU and WinGlob?

Reply:
I just moved this check below, `availableBcpTags` became `collationMapForLocale` because I thought it was a bit clearer.

Reply:
Thanks for clarifying!